src/phoenix/trace/llama_index/callback.py

```diff
@@ -265,17 +292,68 @@ def get_spans(self) -> Iterator[Span]:
         return self._tracer.get_spans()
 
 
-def _add_to_tracer(
-    event_id_to_event_data: Dict[CBEventID, CBEventData],
-    trace_map: Dict[CBEventID, List[CBEventID]],
+def _process_exceptions(
+    event_id_to_event_data: EventMap, trace_map: TraceMap
+) -> Tuple[EventMap, TraceMap]:
```

Comment:
Wouldn't it be simpler to just return a dict as follows? In that case we don't need to mutate/copy anything, and in the final traversal we can just skip the exceptions, instead of having to create a brand new tree excluding the exceptions.
```suggestion
) -> Dict[CBEventID, List[SpanExceptions]]:
```